sql-plugin/src/main/scala/org/apache/spark/sql/rapids/predicates.scala

```diff
@@ -172,6 +172,27 @@ abstract class CudfBinaryComparison extends CudfBinaryOperator with Predicate {
 
   def hasFloatingPointInputs: Boolean = left.dataType == FloatType || left.dataType == DoubleType ||
     right.dataType == FloatType || right.dataType == DoubleType
+
+  def fixFloatingPointResult(
+    result: => ColumnVector,
+    lhsNanPredicate: => BinaryOperable with AutoCloseable,
+    rhsNanPredicate: => BinaryOperable with AutoCloseable
```

Comment:
Nit: The call sites _look like they can leak_ because it's not immediately clear we're deferring the computation of the parameters.  It's not a must-fix from my perspective, but I'm not a fan because of the subtleties this introduces.

Reply:
changing the name to `evaluateAndFixFloatingPointResult` to make it more evident